includes/class-omise-callback.php

```diff
@@ -81,17 +93,6 @@ protected function invalid_result() {
 	 * Resolving a case of charge status: successful.
 	 */
 	protected function payment_successful() {
-		$message = __( 'OMISE: Payment successful.<br/>An amount of %1$s %2$s has been paid', 'omise' );
-
-		$this->order->payment_complete();
-		$this->order->add_order_note(
-			sprintf(
-				wp_kses( $message, array( 'br' => array() ) ),
-				$this->order->get_total(),
-				$this->order->get_order_currency()
-			)
-		);
-
 		WC()->cart->empty_cart();
 		wp_redirect( $this->order->get_checkout_order_received_url() );
```

Comment:
Order status is still failed while redirecting to return page.

  When customer creates order with a failed charge and then create new charge using another payment method, then order get placed normally but it returns to order failure page showing payment error message.

Reply:
@mayurkathale Thanks for testing this case. Can I know the steps you have tested? I wanna try reproduce it.

Reply:
1. Place order using "Alipay".
2. On test payment page, mark payment as failed.
3. Step 2 will take you to checkout page again with failure message error in the notification box.
4. Try to placing same order again with Card Payment with successful charge creation.
5. Step 4 will redirect you to order complete page.
6. Order completed page shows that order placement has been failed.

While backend and omise backend shows payment has done successfully, email gets sent normally.
<img width="918" alt="1" src="https://user-images.githubusercontent.com/5526195/87530063-0e0a6380-c6ba-11ea-8821-4e82a04bfe91.png">

<img width="1248" alt="2" src="https://user-images.githubusercontent.com/5526195/87530174-34300380-c6ba-11ea-8345-483051445135.png">

<img width="822" alt="3" src="https://user-images.githubusercontent.com/5526195/87530197-3a25e480-c6ba-11ea-88cd-8cc405549682.png">

<img width="853" alt="4" src="https://user-images.githubusercontent.com/5526195/87530228-41e58900-c6ba-11ea-9986-e2574f844031.png">

Reply:
Oh, this is interesting case. Let me take a look

Reply:
I couldn't reproduce this case (on both 3-D Secure payment and without).
Can you help test one more time? Maybe it's about browser session cache or something?

Reply:
@guzzilar I have tried in 'incognito mode', I can reproduce this issue. Probably, after step _number 3_, have to make sure that order status is "Failed" and try to execute worker job by doing some activity.

Reply:
ok, I can reproduce the issue now.
Thanks 👍